commit 8fc8c37041dd7845d402ce3a390d0028c7159484
Author: Eva Rachel Retuya <eraretuya@gmail.com>
Date:   Mon Mar 20 19:27:05 2017 +0800

    staging: iio: tsl2x7x_core: Fix standard deviation calculation
    
    commit cf6c77323a96fc40309cc8a4921ef206cccdd961 upstream.
    
    Standard deviation is calculated as the square root of the variance
    where variance is the mean of sample_sum and length. Correct the
    computation of statP->stddev in accordance to the proper calculation.
    
    Fixes: 3c97c08b5735 ("staging: iio: add TAOS tsl2x7x driver")
    Reported-by: Abhiram Balasubramanian <abhiram@cs.utah.edu>
    Signed-off-by: Eva Rachel Retuya <eraretuya@gmail.com>
    Signed-off-by: Jonathan Cameron <jic23@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/iio/light/tsl2x7x_core.c b/drivers/staging/iio/light/tsl2x7x_core.c
index ea15bc1..197201a 100644
--- a/drivers/staging/iio/light/tsl2x7x_core.c
+++ b/drivers/staging/iio/light/tsl2x7x_core.c
@@ -854,7 +854,7 @@ void tsl2x7x_prox_calculate(int *data, int length,
 		tmp = data[i] - statP->mean;
 		sample_sum += tmp * tmp;
 	}
-	statP->stddev = int_sqrt((long)sample_sum) / length;
+	statP->stddev = int_sqrt((long)sample_sum / length);
 }
 
 /**
